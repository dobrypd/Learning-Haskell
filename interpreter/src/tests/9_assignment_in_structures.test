struct {
    int a;
    int b;
} first;

int main() {
    first.a = 10;
    first.b = 20;
    return first.a * first.b;
}
