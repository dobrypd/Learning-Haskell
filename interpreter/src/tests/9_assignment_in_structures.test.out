Welcome to Kappa interpreter
200
[(fromList [(Ident "first",fromList [(Ident "a",10),(Ident "b",20)])],fromList [(Ident "main",([],[SStm (ExprS (Sexpr (Eassign [Ident "first",Ident "a"] (Econst (Eint 10))))),SStm (ExprS (Sexpr (Eassign [Ident "first",Ident "b"] (Econst (Eint 20))))),SStm (JumpS (SjumpFive (Etimes (Evar [Ident "first",Ident "a"]) (Evar [Ident "first",Ident "b"]))))]))],fromList [])]
